app/check_dataset.py

```diff
@@ -187,15 +206,39 @@ def check_history(ds: DataSource) -> ResultLog:
 
     log = ResultLog()
 
+    exceptions_cached = pd.read_csv('./resources/exceptions.csv', parse_dates=['date'])
+
     df = ds.history
     if is_missing(df): 
         log.internal("Source", "History not available")
         return None
 
-    for state in df["state"].drop_duplicates().values:
-        state_df = df.loc[df["state"] == state]
-        checks.monotonically_increasing(state_df, log)
+    for state in df["state"].unique():
+        exceptions = checks.monotonically_increasing(df.loc[df["state"] == state], state, log)
+        if (len(exceptions) > 0):
+            not_cached = list(filter(lambda x: find_in_exceptions(x, exceptions_cached), exceptions))
+            if (len(not_cached) > 0):
+                send_anomaly_info(not_cached)
 
     log.consolidate()
     return log
 
+def cache_exceptions():
```

Comment:
This may belong higher up. The idea is we have a place to go to build the anomaly cache. This would not get run very often I don't think. It duplicates some code in `check_history`

Reply:
I think it belongs a level up and it should call the current/history versions.  anything in those lists are/should be exceptions.